mentoring/mentoring.py

```diff
@@ -83,54 +65,105 @@ class MentoringBlock(XBlockWithLightChildren, StepParentMixin):
-    max_attempts = Integer(help="Number of max attempts for this questions", default=0,
-                           scope=Scope.content, enforce_type=True)
-    mode = String(help="Mode of the mentoring. 'standard' or 'assessment'",
-                  default='standard', scope=Scope.content)
-    step = Integer(help="Keep track of the student assessment progress.",
-                   default=0, scope=Scope.user_state, enforce_type=True)
-    student_results = List(help="Store results of student choices.", default=[],
-                           scope=Scope.user_state)
-
-    display_name = String(help="Display name of the component", default="Mentoring XBlock",
-                          scope=Scope.settings)
+    xml_content = String(help="XML content", default=default_xml_content, scope=Scope.content)
+
+    # Settings
+    weight = Float(
+        help="Defines the maximum total grade of the block.",
+        default=1,
+        scope=Scope.settings,
+        enforce_type=True
+    )
+    display_name = String(
+        help="Display name of the component",
+        default="Mentoring XBlock",
+        scope=Scope.settings
+    )
+
+    # User state
+    attempted = Boolean(
+        help="Has the student attempted this mentoring step?",
+        default=False,
+        scope=Scope.user_state
+    )
+    completed = Boolean(
+        help="Has the student completed this mentoring step?",
+        default=False,
+        scope=Scope.user_state
+    )
+    num_attempts = Integer(
+        help="Number of attempts a user has answered for this questions",
+        default=0,
+        scope=Scope.user_state,
+        enforce_type=True
+    )
+    step = Integer(
+        help="Keep track of the student assessment progress.",
+        default=0,
+        scope=Scope.user_state,
+        enforce_type=True
+    )
+    student_results = List(
+        help="Store results of student choices.",
+        default=[],
+        scope=Scope.user_state
+    )
+
+    # Global user state
+    next_step = String(
+        help="url_name of the next step the student must complete (global to all blocks)",
+        default='mentoring_first',
+        scope=Scope.preferences
+    )
+
     icon_class = 'problem'
     has_score = True
-
-    MENTORING_MODES = ('standard', 'assessment')
+    has_children = True
 
     FLOATING_BLOCKS = (TitleBlock, MentoringMessageBlock, SharedHeaderBlock)
 
-    FIELDS_TO_INIT = ('xml_content',)
-
     @property
     def is_assessment(self):
         return self.mode == 'assessment'
 
     @property
     def score(self):
-        """Compute the student score taking into account the light child weight."""
-        total_child_weight = sum(float(step.weight) for step in self.steps)
+        """Compute the student score taking into account the weight of each step."""
+        weights = [float(self.runtime.get_block(step_id).weight) for step_id in self.steps]
```

Comment:
@e-kolpakov OK I can change, but a list of 10,000 floats should be no problem anyways, right? A list of 10,000 XBlocks would be a problem...